### ISSUE TITLE
Chess - develop logic for move validation, checkmate handling, and all other previously unhandled game elements

### ISSUE THREAD

**Issue by lazarchitect** [lazarchitect/FlaskReactor#70]
We need to add code within the backend server (specifically ChessHandler.py) to handle all previously unhandled elements of chess. These include:

- checkmate + stalemate
- castling
- en passant
- anything else I didn't realize wasn't done yet when I wrote this

we should also be validating these moves on the server side, since theoretically the client could be spoofing requests to send abstract moves, not limited by the UI. Adding this will probably help avoid bugs / race conditions in the future.

A user should be able to play out an entire game of chess, with all mechanics of the game included, and finish a game.

**Comment by lazarchitect** [comment]
Since this is a huge part of FlaskReactor's features, it's understandable that almost all the major Labels would be used. This is also the first story to incorporate the Milestone feature, hoping to release v0.6.0 with this change soon.

**Comment by lazarchitect** [comment]
I'm now realizing this should probably be broken up into a few items. Castling by itself isn't too big of a feature to add, but coding and testing it by itself would be a small-to-medium sized ticket; all of these features together feel more like an epic where multiple tested items come together.
'

**Comment by lazarchitect** [comment]
Castling needs to be managed via react useState. We can pull *whiteKingMoved* and *blackKingMoved* from the payload, but those fields should also be managed by react and updated after a new websocket message comes in from the server, in the exact same way that *boardstate* is managed and updated each time. The only difference is that these two fields are not a visible UI element.

**Comment by lazarchitect** [comment]
confirmed - React NOT needed for kingMoved fields, we can just track it globally in the browser same as **yourTurn** or **highlightedTiles**

**Comment by lazarchitect** [comment]
After a bit of delay to collect my thoughts and write them down in the Karst notebook, I've decided to go ahead with divvying up the workload of this issue into several smaller issues. There's going to be at least 4 stories, one for each item (castling, checkmate, stalemate, and en passant). I will probably create other stories for the underlying subroutines that support these features, such as determining move legality, tile danger status, etc. Stay tuned for stories that will be marked as dependencies for this one.

**Comment by lazarchitect** [comment]
Closing due to the creation of other, more granular issues which will cover the MVP scope of this issue for the 0.6.0 release. More details to come in 0.7.0 items for any non-MVP work.

**Comment by lazarchitect** [comment]
Just coming back here after #84 and chuckling at "React NOT needed for kingMoved fields, we can just track it globally". After dividing up chessboard.js into meaningful submodules, this would not be a good approach - and using React WOULD be a good approach. So, yeah. Learnings.
